agent/host_agent_test.go

```diff
@@ -861,4 +862,80 @@ users:
 			Expect(err.Error()).Should(ContainSubstring("kubeconfig generation failed: hostname is not valid"))
 		})
 	})
+	Context("When the certRotation func is called", func() {
+		var (
+			bootstrapKubeConf *os.File
+			err               error
+		)
+		BeforeEach(func() {
+			bootstrapKubeConf, err = ioutil.TempFile("", "bootstrap-kubeconfig")
+			Expect(err).NotTo(HaveOccurred())
+			bootstrapKubeConfig = bootstrapKubeConf.Name()
+		})
+		AfterEach(func() {
+			Expect(os.Remove(bootstrapKubeConf.Name())).ShouldNot(HaveOccurred())
+		})
+		It("should return if certificate data is not valid", func() {
```

Comment:
```suggestion
		It("should return error if certificate data is not valid", func() {
```

Reply:
Existing code does not return error if certificate data is not valid (it just logs the failures). So this test also checks that the method should return without error.